src/abstract_block.jl

```diff
@@ -363,3 +370,37 @@ function parameters_range!(out::Vector{Tuple{T,T}}, block::AbstractBlock) where
         parameters_range!(out, subblock)
     end
 end
+
+# non-inplace versions
+"""
+    apply(register, block)
+
+The non-inplace version of applying a block (of quantum circuit) to a quantum register.
+Check `apply!` for the faster inplace version.
+"""
+apply(r::AbstractRegister, b) = apply!(copy(r), b)
+
+function generic_dispatch!(f::Union{Function,Nothing}, x::AbstractBlock, it::Dispatcher)
```

Comment:
why do we have this function now? and this function seems to be only used by `dispatch`?

Reply:
It is not exported and it is just for realizing `dispatch`.